Plugin/CatalogGraphQl/Products/Query/Search.php

```diff
@@ -98,29 +98,16 @@ public function afterGetResult(
         SearchResult  $searchResult,
         array $args
     ) {
-        if (isset($args[self::SORT_KEY]) && isset($args[self::SORT_KEY][CategorySorting::NOSTO_PERSONALIZED_KEY])
-            && $this->getTotalPages() !== null) {
+        if (isset($args[self::SORT_KEY]) && isset($args[self::SORT_KEY][CategorySorting::NOSTO_PERSONALIZED_KEY])) {
             return $this->searchResultFactory->create([
                 'totalCount' => $searchResult->getTotalCount(),
                 'productsSearchResult' => $searchResult->getProductsSearchResult(),
                 'searchAggregation' => $searchResult->getSearchAggregation(),
                 'pageSize' => $searchResult->getPageSize(),
                 'currentPage' => $searchResult->getCurrentPage(),
-                'totalPages' => $this->getTotalPages(),
+                'totalPages' => $searchResult->getTotalCount(),
```

Comment:
We need to alter the nr of total products based on the number of products on Nosto's merchandise.